README.md

```diff
@@ -1,11 +1,20 @@
-A fork of [Go tools](golang.org/x/tools).
+# go-diff-patch 
 
-Experimenting.
+[![Go CI](https://github.com/sourcegraph/go-diff-patch/actions/workflows/go-ci.yml/badge.svg)](https://github.com/sourcegraph/go-diff-patch/actions/workflows/go-ci.yml)
+[![lsif-go](https://github.com/sourcegraph/go-diff-patch/actions/workflows/lsif-go.yml/badge.svg)](https://github.com/sourcegraph/go-diff-patch/actions/workflows/lsif-go.yml)
 
-To test stuff, run the command
+A fork of [Go tools](golang.org/x/tools). Go-diff-patch is an utility library that is used to generate git-compatible patches that are appliable to any repository.
+
+## Installation
+
+`go-diff-patch` can be installed using the `go get` command as shown below:
 
 ```sh
-go run main.go
+go get github.com/sourcegraph/go-diff-patch
 ```
 
-A `generated.patch` file should be generated and work is still ongoing to make it `git-compatible` so it can be applied with `git apply <path/to/patch>`
\ No newline at end of file
+### Usage
+
+The library exports a function `GeneratePatch` which returns the different between two files in a string format. This patch can then be applied to a valid git repository containing the specified file(s).
```

Comment:
Do we intend to only return the diff in a string format?

Maybe we could make the inputs and outputs clearer:

```suggestion
The library exports a function `GeneratePatch` which computes edits between the original and updated file contents and produces a unified diff for them as a string. This patch can then be applied to a valid git repository containing the specified file(s).
```

Reply:
Yes, at this moment that's all we do., however in the future we can easily extend this to any other format we like.